syntax = "proto3";

package notes;

message Note {
  string id = 1;
  string title = 2;
  string description = 3;
  string created_at = 4;
  string updated_at = 5;
}

message CreateNoteRequest {
  string title = 1;
  string description = 2;
}

message CreateNoteResponse {
  Note note = 1;
}

message GetNoteRequest {
  string id = 1;
}

message GetNoteResponse {
  Note note = 1;
}

message ListNotesRequest {}

message ListNotesResponse {
  repeated Note notes = 1;
}

message UpdateNoteDescriptionRequest {
  string id = 1;
  string description = 2;
}

message UpdateNoteDescriptionResponse {
  Note note = 1;
}

message DeleteNoteRequest {
  string id = 1;
}

message DeleteNoteResponse {}

message HealthRequest {}
message HealthResponse {}

service NotesService {
  rpc CreateNote(CreateNoteRequest) returns (CreateNoteResponse);
  rpc GetNote(GetNoteRequest) returns (GetNoteResponse);
  rpc ListNotes(ListNotesRequest) returns (ListNotesResponse);
  rpc UpdateNoteDescription(UpdateNoteDescriptionRequest)
      returns (UpdateNoteDescriptionResponse);
  rpc DeleteNote(DeleteNoteRequest) returns (DeleteNoteResponse);
  rpc Health(HealthRequest) returns (HealthResponse);
}
